collections/_posts/2024-02-23-Rubiks-cube.md

```diff
@@ -0,0 +1,87 @@
+
+![](/collections/images/3D_Rubik/method_diagram.jpg)
+
+The method is inspired by jigsaw puzzle pretext task but adapted for 3D medical images with an increased difficulty by adding some rotation.
+The first step is to separate the volume into cubes, which are then rearranged and randomly rotated. Since the goal is to learn high-level semantic features and not texture information close to cubes boundaries, a gap of "about 10 voxels" is left between two adjacent cubes. Each cube is also normalized to [-1,1].
+
+## Network architecture
+
+$$M$$ siamese networks ($$M$$ the number of cubes) with shared weight branches, that they call Siamese-Octad, are used to solve the problem. The backbone can be any 3D CNN. The feature maps of the last layer of all branches are concatenated and given to a last fully connected layer for the too tasks and supervised by permutation loss and rotation loss.
+
+## Cube ordering
+
+As there are $$M!$$ permutations and that some may be very close to each other, too ambiguous and challenging to distinguish, they iteratively select $$K$$ permutations with the largest Hamming distance. The network is then trained to recognize which one of the $$K$$ permutations has been applied to the input volume: it is a classification task with $$K$$ categories. With $$p$$ the network prediction and $$l$$ the one-hot label, the loss is defined as 
+
+$$ L_p = - \sum{l_j \log p_j}$$
+
+## Cube orientation
+
+In order not to extract only translation invariant features, they add some rotation. For each cube there could be 3 (axes) * 2 (directions) * 4 (angles) = 24 configurations. Again, to reduce the complexity, they limited the modifications to 180° horizontal and vertical rotations. The network has to recognize whether each cube has been rotated or not: it is a multi-label classification task with two $$1*M$$ vectors for each direction with 1 at the position of rotated cubes and 0 otherwise as ground truth. If $$r$$ are the predicted vectors and $$g$$ the ground truth, the loss is
+
+$$ L_r = - \sum_i^M{g_i^{hor} \log r_i^{hor} + g_i^{ver} \log r_i^{ver}}$$
+
+## Objective
+
+The global loss is weighted according to the importance of the two tasks. 
+
+$$L = \alpha L_p + \beta L_r$$
+
+They experimentaly found that $$\alpha = \beta = 0.5$$ are the best weights.
+
+## Weights transfer
+
+For a target classification task, the pretrained CNN can be directly fine-tuned but for a segmentation task, the weights can only be transferred to the encoder of a FCN. As the random initialization of the decoder might neutralize the improvement brought by the pretraining, they apply convolutional operations directly on the feature maps provided by the pre-trained encoder to obtain the dense pixel-wise prediction instead of the transposed convolutions, inspired by dense upsampling convolutions which has fewer trainable parameters in the decoder.
```

Comment:
The last sentence is a bit long and complex in my opinion. I'd split it up at least a little bit, but feel free to suggest another way to rephrase things!
```suggestion
For a target classification task, the pretrained CNN can be directly fine-tuned but for a segmentation task, the weights can only be transferred to the encoder of a FCN. As the random initialization of the decoder might neutralize the improvement brought by the pretraining, they apply convolutional operations directly on the feature maps of the pre-trained encoder to obtain dense pixel-wise prediction, instead of using transposed convolutions on downsampled feature maps (i.e. traditional decoder architectures). This decoder architecture was inspired by dense upsampling convolutions, which has fewer trainable parameters in the decoder.
```